<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<meta http-equiv="Cache-Control" content="no-cache, no-store, must-revalidate">
<meta http-equiv="Pragma" content="no-cache">
<meta http-equiv="Expires" content="0"><meta http-equiv="Content-Language" content="en-us">
<meta http-equiv="Content-Type" content=
"text/html; charset=us-ascii">
<title>muscle manual</title>
<link rel="stylesheet" type="text/css" href="manstyles.css">
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-50117757-1', 'auto');
  ga('send', 'pageview');

</script>

</head>
<body bgcolor="#E1E1E1">
<div align="center">&nbsp;
<table border="0" width="735" cellspacing="0" cellpadding="5"
bgcolor="#FFFFFF">
<tr>
<td bgcolor="#E1E1E1"><span class="ManPath"><a href=
 "index.html">MUSCLE manual</a></span></td>
</tr>
<tr>
<td bgcolor="#006600"><span class="ManTopic">Diagonal optimization</span></td>
</tr>
<tr>
<td>
<span class="ManText">&nbsp;<br>
Creating a pair-wise alignment by dynamic programming requires computing an N x 
M matrix, where N and M are the sequence lengths. A trick used in algorithms 
such as BLAST is to reduce the size of this matrix by using fast methods to find 
&quot;diagonals&quot;, also called &quot;gapless high-scoring segment pairs&quot;, i.e. short 
regions of high similarity between the two sequences. This speeds up the 
algorithm at the expense of some reduction in accuracy. MUSCLE uses a technique 
called k-mer extension to find diagonals, which is described in this paper:<br>
<br>
Edgar, R.C. (2004) Local homology recognition and distance measures in linear 
time using compressed amino acid alphabets, Nucleic Acids Res., 32, 1. [<a href="http://www.ncbi.nlm.nih.gov/pubmed?term=14729922">Pubmed</a>].<br>
<br>
It is disabled by default because of the slight reduction in average accuracy, 
and can be turned on by specifying the &#8211;diags option. To enable diagonal 
optimization only in the first iteration, use &#8211;diags1, to enable diagonal 
optimization in the second iteration only use &#8211;diags2. These are provided separately 
because it would be a reasonable strategy to enable diagonals in the first 
iteration but not the second (because the main goal of the first iteration is to 
construct a multiple alignment quickly in order to improve the distance matrix, 
which is not very sensitive to alignment quality; whereas the goal of the second 
iteration is to make the best possible progressive alignment).<br>
&nbsp; </span></td>
</tr>
</table>
</div>
</body>
</html>
