<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<meta http-equiv="Cache-Control" content="no-cache, no-store, must-revalidate">
<meta http-equiv="Pragma" content="no-cache">
<meta http-equiv="Expires" content="0"><meta http-equiv="Content-Language" content="en-us">
<meta http-equiv="Content-Type" content=
"text/html; charset=us-ascii">
<title>muscle manual</title>
<link rel="stylesheet" type="text/css" href="manstyles.css">
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-50117757-1', 'auto');
  ga('send', 'pageview');

</script>

</head>
<body bgcolor="#E1E1E1">
<div align="center">&nbsp;
<table border="0" width="735" cellspacing="0" cellpadding="5"
bgcolor="#FFFFFF">
<tr>
<td bgcolor="#E1E1E1"><span class="ManPath"><a href=
 "index.html">MUSCLE manual</a></span></td>
</tr>
<tr>
<td bgcolor="#006600"><span class="ManTopic">Making a Neighbor-Joining or UPGMA 
tree</span></td>
</tr>
<tr>
<td>
<p class="Code"><span class="ManText">&nbsp;<br>
Make a <a href="http://en.wikipedia.org/wiki/UPGMA">UPGMA</a> tree from a 
multiple alignment:</span></p>
<p class="Code"><span class="ManCode">&nbsp;muscle -maketree -in seqs.afa -out seqs.phy</span></p>
<p class="MsoNormal"><span class="ManText">Make a
<a href="http://en.wikipedia.org/wiki/Neighbor_joining">Neighbor-Joining</a> 
tree from a multiple alignment:</span></p>
<p class="Code"><span class="ManCode">&nbsp; muscle -maketree -in seqs.afa -out seqs.phy 
-cluster neighborjoining</span></p>
<p class="Code"><span class="ManText"><b>UPGMA or Neighbor joining?<br>
</b>Neighbor-joining trees will usually be a better estimate of the true 
phylogenetic tree. UPGMA is faster, which can be useful with large datasets when 
N-J is too slow.</span></p>
<p class="MsoNormal"><span class="ManText"><b>Input file<br>
</b>Specified using the -in option. Must be in aligned <a href="fasta_files.html">FASTA 
format</a>.</span></p>
<p class="MsoNormal"><span class="ManText"><b>Output file<br>
</b>Specified using the -out option. The tree is written in
<a href="http://en.wikipedia.org/wiki/Newick_format">Newick format</a>, which is 
supported by most phylogenetic analysis packages such as
<a href="http://evolution.genetics.washington.edu/phylip.html">PHYLIP</a>. See 
also this
<a href="http://evolution.genetics.washington.edu/phylip/newicktree.html">
article on the Newick format</a> in the PHYLIP documentation.<br>
&nbsp;</span></p></td>
</tr>
</table>
</div>
</body>
</html>
